apps/docs/js-interpreter.md

```diff
@@ -0,0 +1,167 @@
+# Upgrading the JS Interpreter
+
+## Component Overview
+
+Before embarking on an upgrade of the JS interpreter,
+you need to know what all the different little pieces
+are and how they relate to each other.
+
+### Neil Fraser's JS-Interpreter Repository
+
+[NeilFraser/JS-Intepreter](https://github.com/NeilFraser/JS-Interpreter)
+is the main repository where development of the interpreter takes place. It gets updated
+sporadically every 3-6 months or so. The goal of "upgrading the interpreter" is about
+getting the updates in this repository to work on code.org.
+
+### Code.org's "fork" of Neil's JS-Interpreter
```

Comment:
Interesting.  That makes sense, but I'm not sure it has exactly that connotation in the context of GitHub in particular.

> **Fork**
> 
> A fork is a personal copy of another user's repository that lives on your account. Forks allow you to freely make changes to a project without affecting the original. Forks remain attached to the original, allowing you to submit a pull request to the original's author to update with your changes. You can also keep your fork up to date by pulling in updates from the original.
> 
> From the _[GitHub Glossary](https://help.github.com/articles/github-glossary/#fork)_